core/modules/url/pages/UrlClientDispatcher.vue

```diff
@@ -0,0 +1,76 @@
+<template>
+  <component :is="page" />
+</template>
+
+<script>
+import { UrlDispatchMapper, _handleDispatcherNotFound } from '@vue-storefront/core/modules/url'
+import userRoutes from 'theme/router'
+import store from '@vue-storefront/store'
+import { AsyncDataLoader } from '@vue-storefront/core/lib/async-data-loader'
+
+let _matchingComponentInstance = null
+const dispatcherRoutine = (to, from, next) => {
+  UrlDispatchMapper(to).then(routeData => {
+    if (routeData) {
+      if (store.state.config.seo.useUrlDispatcher && routeData) {
+        const userRoute = userRoutes.find(r => r.name === routeData['name'])
+        if (userRoute) {
+          userRoute.component().then(rootComponent => {
+            _matchingComponentInstance = rootComponent.default
+
+            if (from.name) { // don't execute the action on the Client's side of the SSR actions
+              if (_matchingComponentInstance.asyncData) {
+                AsyncDataLoader.push({
```

Comment:
AsyncDataLoader will be removed in next version, instead update to vue 2.6 and use native nehavior